### DISCUSSION TITLE
Watch stats

### DISCUSSION THREAD

**Discussion by bartoszek** [ccache/ccache#1008] [Ideas]
I have this "smallish" script that shows cumulative cache statistics counting since it's started, but it works only with `ccache<=4.3`
It also tries to calculate the number of files and size of evicted cache entries.
http://ix.io/3QhD/bash
Wonder if someone implements any such tool, or perhaps it could be integrated into `--stat-view` switch with something like `--stat-view-interval, --stat-view-cumulative` option.
_side notice regarding stats: hit rate in `cache -s` is not rounded correctly :> 75.995 is displayed as 75.99 instead of 76.00_

**Comment by jrosdahl** [comment]
> side notice regarding stats: hit rate in cache -s is not rounded correctly :> 75.995 is displayed as 75.99 instead of 76.00

I can't reproduce that. Ccache uses `https://fmt.dev` with format string `{:.2f}` to print percentages. It behaves like this in my test program:

    fmt::print("75.9949 -> {:.2f}\n", 75.9949); // prints 75.9949 -> 75.99
    fmt::print("75.9950 -> {:.2f}\n", 75.9950); // prints 75.9950 -> 76.00
    fmt::print("75.9951 -> {:.2f}\n", 75.9951); // prints 75.9951 -> 76.00
